src/app/services/project.service.ts

```diff
@@ -48,25 +48,69 @@ export class ProjectService {
   }
 
   private getCollections() {
-    return this.api.getProjectCollections(this.project.name.replace(/\W+/g, '-').toLowerCase()).pipe(
-      map((res: any) => this.processCollections(res)),
-      map((collections: any[]) => {
+    return this.api.getProjectCollections(this.project._id).pipe(
+      map((res: any) => this.processCollections(res && res[0] && res[0].searchResults ? res[0].searchResults : null)),
+      map(async (collections: any[]) => {
         // Push them into the project
-        collections.forEach(collection => {
+        await this.loadCollectionRecords(collections);
+
+        collections.filter(c => c.documents.length > 0).forEach(collection => {
           this.addCollection(this.project.collections, collection);
         });
-      }));
+      })
+    );
   }
 
   private processCollections(res: any[]): any[] {
     const collections = res ? res : [];
 
     collections.forEach((collection, index) => {
-      collections[index] = new Collection(this.api.hostnameNRPTI, this.api.hostnameNRPTI, collection);
+      collections[index] = new Collection(collection);
     });
+
     return collections;
   }
 
+  private async loadCollectionRecords(collectionsList: any[]) {
+    for (const collection of collectionsList) {
+      for (const recordId of collection.records) {
+        // fetch the record from NRPTI
+        const recordResult = await this.api.getCollectionRecord(recordId).toPromise();
+        const loadedRecord = recordResult && recordResult['length'] > 0 ? recordResult[0] : null;
```

Comment:
Small nit: don't normally access an array's properties with a key. Normally `recordResult.length`.

Reply:
Normally, but it's a typescript compile issue. It's using `Object` type, rather than `any` type, so will throw `Property 'length' does not exist on type 'Object'.ts(2339)`.  Hence the key :(